src/plugins/dashboard/public/dashboard_container/component/grid/dashboard_grid.tsx

```diff
@@ -125,8 +125,7 @@ export const DashboardGrid = ({ viewportWidth }: { viewportWidth: number }) => {
       className={classes}
       width={viewportWidth}
       breakpoints={breakpoints}
-      onDragStop={onLayoutChange}
-      onResizeStop={onLayoutChange}
```

Comment:
@ThomThomson There is a [functional test failure](https://buildkiteartifacts.com/e0f3970e-3a75-4621-919f-e6c773e2bb12/0fda5127-f57f-42fb-8e5a-146b3d535916/018ac220-1baa-4f45-bb9e-0b46415a2c6f/018ac241-c1ea-4de0-a84a-08441e753054/target/test_failures/018ac241-c1ea-4de0-a84a-08441e753054_7670b1d71a6bc6d6025a44f1dbf7c11d.html?response-content-type=text%2Fhtml&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7LX7YFRQ5T%2F20230923%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230923T175911Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEDgaCXVzLWVhc3QtMSJIMEYCIQDhkPW6Ez2%2Bgc01aRwkRLBdEDOCtZgDEDTwKf5Do4C1MwIhAM5pF5X2PHCwzwsRZXBWePYU4uLLsfvYox5VkH%2BY4UU9KvEDCDAQABoMMDMyMzc5NzA1MzAzIgzlK107yLaph8nlwZMqzgMLaESWahd%2FvzAQjU6vjO8Lyl3Pgfkwb0xE2oGS3WTOMBmEKfhvPxuD%2FJJ8EbOkYUwDd7mlhTiE7M1BDPZVSOpBN%2FZSMmRTxZJHjZyls6Y%2Fo%2Bfj4MmEXzlbJFeXW3jyNGRaWwzw4xebHuerINwqpWWR22ORCFJ2t9H3mGSS6LHb4upJjdFF%2Bu%2BJZqHB0w5AKF%2Fv%2BaZIn7Nr5qM5gGVuwHy82J60IgCl4geoKX93NTUCJZohcJ16bYf8JDUDT297Ym8QS08Bym3LdYB4ry0T9Cy5TKAxr1xoCD5znPTjOL2Bt9V4e2nwu3BKpdGMaaXdCNdwYBGdrt0xkLaIqLBwDYm17TnTX04mMJ%2FEUa%2BicpTwfar1Oyx5WoQCF6HqcDQqEJKyTztm%2BGTDC1GImbeIfV7YejjSB7qSEppOAFGQbNLeegZ%2BK%2FAB48gPMtaviiltvBpWit6yRe3QnmBumPLjTeeLjWJyHArG1lntwKAU2m6MnztEOjaTPP51gLBgt5DyOOSYMMgDkQwMuUPS2iwIWpYDJ%2FeGdz0I2I7p7Y9GImIdePXhqFYvuTg2yHvGmU23Y6Eh5bU5hYJ6OxDag3Xm8OVdYSSTOSe1mK0JdOwrvFsw3P27qAY6pAHdpIEeVz7tsul86RmDLoh1qHuSJUomPoP6jubFEqPpmei%2FjI1LdaNPGkjErZ6NUpRIud10kIhO0yWW3deOHRAnjVtK9mjR8hW7mcOYsRbZ2lXgJ00oHodVpvL3hZ%2F62PTQ8r3QRx%2FepONoOw4ca6Y2XMqu0mll8L%2BX8vYVuag9gybgRAdo1XqdYQz9y6EKzuO0auYjmo%2BpSjys6ko%2FdKsVznOkpw%3D%3D&X-Amz-SignedHeaders=host&X-Amz-Signature=934275a22ba012b70175410336164bae663c5ee7218a2f9e531345ec7ca46f2f) that appears to be related to this change. The failing test is in test/functional/apps/dashboard/group1/url_field_formatter.ts and called `applied on dashboard`.

I did some debugging and compared to the test in `main`. Notice how the screenshot shows the panel to be very small at the edge of the screen? In `main` the panel is sized appropriately to see all the data.

In my local tests, I reverted the change here and the test passes. But that might break something elsewhere. Can you look into it? I'm also happy to join a sync to debug together.

Reply:
This is so strange. I changed these out because with only `onDragStop` and `onResizeStop`, the panels were only getting recalculated on manual interactions, therefore they would get out of sync when the grid was changed programatically (e.g. during a clone panel).

This makes me wonder if there are some cases where `onDragStop` or `onResizeStop` fire events that the change handler doesn't fire. Have you tried using all 3?

Reply:
@ThomThomson I did some more debugging. The problem stems from the [`grid.y` property](https://github.com/elastic/kibana/blob/main/test/functional/fixtures/kbn_archiver/dashboard/current/kibana.json#L2291) for the panel in the dashboard used for the functional test. In this case, `grid.y = 15` instead of `0` which should normally be the case for a single panel dashboard. I'm not sure how this dashboard saved object was created with a `y: 15`? 

Ultimately, this messes with the ReactGridLayout which receives a `viewport` prop of 1. So the `viewport` is 1 and now `onLayoutChange` is called with `newLayout.grid.w: 1` when the dashboard is rendered. 

Here is a saved object that you can import to see it in action. It has two dashboards, "good y" and "bad y". The dashboards are identical, except for the `grid.y` of the panel. 
[dashboard_with_bad_y.ndjson.zip](https://github.com/elastic/kibana/files/12718813/dashboard_with_bad_y.ndjson.zip)

Do you know if older versions of Kibana could have created a panel on the top row with a `grid.y` value that is greater than zero?

Reply:
I'm pretty certain that there was never a way to save a Dashboard with a panel with a y of 15 without any panels above it. Likely what happened here is that the Dashboard was exported and then manually edited after the fact.

When I test the good y / bad y Dashboards in a branch with the `onLayoutChange` callback, I'm only seeing that the `bad Y` has unsaved changes (the panel location changed). Is that what you're seeing?

If so, I'd say the best course of action here is to simply update the saved object in the test.

Reply:
Upon further review, I suspect this panel was copied from [another saved object](https://github.com/elastic/kibana/blob/main/test/functional/fixtures/kbn_archiver/dashboard/current/kibana.json#L542C1-L542C485) which would explain why `grid.y = 15`. 😞